dot/network/stream_manager.go

```diff
@@ -51,18 +52,18 @@ func (sm *streamManager) start() {
 }
 
 func (sm *streamManager) cleanupStreams() {
-	sm.streamDataMap.Range(func(id, data interface{}) bool {
-		sdata := data.(*streamData)
-		lastReceived := sdata.lastReceivedMessage
-		stream := sdata.stream
+	sm.streamDataMu.Lock()
+	defer sm.streamDataMu.Unlock()
+
+	for id, data := range sm.streamData {
+		lastReceived := data.lastReceivedMessage
+		stream := data.stream
 
 		if time.Since(lastReceived) > cleanupStreamInterval {
 			_ = stream.Close()
```

Comment:
as `stream.Close` returns an error, can we log this error
```suggestion
			err := stream.Close()
			if err != nil {
			    logger.Wanrf("failed to close stream: %w", err)
			}
```

Reply:
Out of scope in this PR (also, not going to lie, all the CI passed :smile:)

On the other hand, this close error might happen regularly too so we may not want to log it (at least no as info/warn/error). Anyway, sorry I'll ignore this and mergggge but I would had totally took a look if the CI would had failed or would be more stable :smile: